content/shmem_sync_all.tex

```diff
@@ -0,0 +1,44 @@
+\color{ForestGreen}
+\apisummary{
+    Registers the arrival of a \ac{PE} at a barrier and suspends \ac{PE}
```

Comment:
From this PR perspective, at least for me, it seems more appropriate to use “block the calling thread”.  When you make a call to sync, all non-blocking operations might be still progressing, is that a correct behavior, when you “suspend the execution”.

hmm .. I don’t recall the discussion. However, I will not defend the use of “suspends execution” in barrier. :)